### PR TITLE
Codecov github actions

### PR DESCRIPTION
**Pull request by v-rocheleau**
Codecov's pypi package was [deprecated](https://about.codecov.io/blog/message-regarding-the-pypi-package/#:~:text=On%20April%2012%2C%202023%2C%20Codecov,used%20package%20from%20active%20support.&text=As%20this%20package%20was%20unmaintained,to%20be%20made%20functional%20again) last week.

This PR removes the dependency on the old codecov python package, and replaces it with a Github Action step.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -57,7 +57,6 @@ tabulate = "^0.9.0"
 uritemplate = "^4.1.1"
 
 [tool.poetry.group.dev.dependencies]
-codecov = "^2.1.12"
```
Comment (davidlougheed) [inline pyproject.toml:60]: this needs a corresponding (should be automated by poetry) removal from the lockfile

**Review by davidlougheed — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (v-rocheleau) [reply pyproject.toml:60]: Good catch, didn't know I had to run `poetry lock` for it to update

**Review by davidlougheed — APPROVED** [review]
lgtm

File: pyproject.toml (hunk 1 above)
Comment (davidlougheed) [reply pyproject.toml:60]: there are commands that remove an entry from pyproject.toml and poetry.lock at once (see poetry docs), but if you remove an item by hand you need the lock command yeah

**Pull request merged**